src/julia-parser.scm

```diff
@@ -809,8 +809,8 @@
           (else ex))))
 
 (define (parse-arrow s) (parse-RtoL s parse-or         is-prec-arrow? (eq? t '-->) parse-arrow))
-(define (parse-or s)    (parse-RtoL s parse-and        is-prec-lazy-or? #t parse-or))
-(define (parse-and s)   (parse-RtoL s parse-comparison is-prec-lazy-and? #t parse-and))
+(define (parse-or s)    (parse-RtoL s parse-and        is-prec-lazy-or? (eq? t '|\|\||) parse-or))
+(define (parse-and s)   (parse-RtoL s parse-comparison is-prec-lazy-and? (eq? t '&&) parse-and))
```

Comment:
I think it would make sense for `.||` and `.&&` to be syntactic operators as well instead of `call` expressions, since they don't really behave like regular function calls.

Reply:
Yeah, my first attempt used special syntax forms in the `expand-table`:

```lisp
   '.&&
   (lambda (e) (expand-fuse-broadcast '() `(|.| (top andand) (tuple ,@(cdr e)))))
   '|.\|\||
   (lambda (e) (expand-fuse-broadcast '() `(|.| (top oror) (tuple ,@(cdr e)))))
```

This, however, doesn't do the _expand_ part with siblings because _all_ the machinery in `expand-fuse-broadcast` only works with `call` heads.  That is, chains like `a .&& b .&& c` end up with _two_ `materializes` — two separate fused loops.  I don't have the lispy chops to do much more than this.

Reply:
I think the problem is just this line https://github.com/JuliaLang/julia/blob/9e5b423a9bbbf70cffa602a4d71d7e0a0dee222b/src/julia-syntax.scm#L1801. It shouldn't be hard to make `.||` expression heads work here as well.

Reply:
It seemed like a lot of that code has pretty strong assumptions about, e.g., `f` being `(cadr e)` and such.  Looks like I forgot to remove the `expand-table` definitions for `.||` heads in here — I'd really be just as happy for it to work that way.  Please feel free to push directly to this branch.

Reply:
(See also #39600 which is the same problem)

Reply:
I am going to wait until #39602 is merged, since I don't feel like rebasing this twice :)